src/NextGen/DonationForm/resources/types.ts

```diff
@@ -82,29 +100,43 @@ export interface Gateway {
     afterCreatePayment?(response: object): Promise<void> | Error;
 }
 
-export interface FormDesign {
-    id: string;
-    fields?: {
-        amount?: ReactNode;
-        text?: ReactNode;
-        textarea?: ReactNode;
-        email?: ReactNode;
-        hidden?: ReactNode;
-    };
-    elements?: {
-        html?: ReactNode;
-    };
-    groups?: {
-        name?: ReactNode;
-    };
-    layouts?: {
-        section?: ReactNode;
-        form?: ReactNode;
-        header?: ReactNode;
-        title?: ReactNode;
-        description?: ReactNode;
-        goal?: ReactNode;
-    };
+export interface FormTemplateFields {
+    amount: FC<AmountProps>;
+    text: FC<FieldProps>;
+    textarea: FC<FieldProps>;
+    email: FC<FieldProps>;
+    hidden: FC<FieldProps>;
+    gateways: FC<GatewayFieldProps>;
+    select: FC<SelectFieldProps>;
+}
```

Comment:
Let's import the `defautlFormTemplates` and use that to provide the specific types, then add a dynamic type for the ones added later. Here's an [example of what I mean](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbwGIGEC+cBmUIjgIigFMBDAYxnwFgAoWgEyLIBsTi4yIA7AZ3gBUiADxgAuOKgA8JLgE8AfAyat2nXgOEwAgsRLipMhbVpq+cRphIBXZjCTQQg8KxhEecALyJacX1mBEzPQ84gg+fhGuIuKCIgA04RG+UTBspDGaOqSJcGi0eXQ0MLJgRBIOTmAuREgBQe5exaUQmOZEljZ2FUTOJK48AHSYdcFwAGRwAEpM0PSSfFDAXADmcRIo0nLyioWMLGkc3GbTZFZQiyv6G4Y7Jkfww4HB+t29rrVPDd40kZoZ8TkUmk9HBYtpdAkfn5iKdzktluITmcLst8kA). The same can be done for elements and groups.

As a note, the `import type` declarative can be used on things not just types (in this case defaultFormTemplates). It just tells it not to actually import it when bundling.

Reply:
I see what your saying, just let the `defaultFormTemplate` dictate what the FormTemplates type is.  I decided to simplify this whole thing here b7c7a4d3 because defining the types is kind of pointless now, since the only place it was being used was for the initial object.

I'm actually already using the typeof the templates from the window for the template wrapper too so everything just stems from that initial `defaultFormTemplate` which makes things easier.